code/script.js

```diff
@@ -34,18 +43,282 @@ const showMessage = (message, sender) => {
 }
 
 // Starts here
+// Bot greets user
 const greetUser = () => {
   // here we call the function showMessage, that we declared earlier with the argument "Hello there, What's your name?" for message, and the argument "bot" for sender
-  showMessage("Hello there, What's your name?", 'bot')
+  showMessage("Hi! I can help you find the perfect gift for anyone. <br><br>Please start by telling me your name.", 'bot')
   // Just to check it out, change 'bot' to 'user' here 👆
 }
 
+//The user types name and name is saved
+const handleNameInput = (event) => { 
+  event.preventDefault(); 
+  //name saved for later
+  name = nameInput.value
+  console.log(name)
+  //the users name pops up as a message
+  showMessage(name, "user")
+  //name input is reset
+  nameInput.value = ""; 
+  //after 1s bot asks for gender
+  setTimeout(() => askForGender(name), 1000);
+}
+
+const askForGender = (name) => {
+  //Bot's welcome message with name pops up. Asks for gender.
+  showMessage(`Nice to meet you ${name}! <br><br> Are you looking for a gift for a gentleman or a lady?`, "bot")
+  //Buttons with gender choice appears
+  inputWrapper.innerHTML =`
+  <button class="gender-btn" id="lady-btn">👩‍🦰</button>
+  <button class="gender-btn" id="gentleman-btn">🧔</button>
+  `; 
+  //Adding eventlisteners to gender buttons
+  document.getElementById('lady-btn')
+    .addEventListener('click', () => {
+      //redeclaring global variable "gender"
+      gender = "lady"
+      //User reply comes up
+      showMessage("It's for a lady.", "user")
+      //Invokes bot's next message
+      setTimeout(() => confirmGender(gender), 1000)
+    })
+  document.getElementById('gentleman-btn')
+    .addEventListener('click', () => {
+      //redeclaring global variable "gender"
+      gender = "gentleman"
+      //User reply comes up
+      showMessage("It's for a gentleman.", "user")
+      //Invokes bot's next message
+      setTimeout(() => confirmGender (gender), 1000)
```

Comment:
Great use of parameters and arguments in these functions - lovely! I would just remove the little space between `confirmGender` and the parenthesis with the parameter `(gender)`, but that's really not a big thing 🤭